File: WorldFeed/Server/src/WorldFeed.Common/Models/WebhookEnvironment.cs
namespace WorldFeed.Common.Models
{
    using System.Linq;
    using Newtonsoft.Json;

    using global::WorldFeed.Common.Public;
    using global::WorldFeed.Common.Public.Models.Interfaces;
    using global::WorldFeed.Common.Public.Models.Interfaces.DTO.Webhooks;

    public class WebhookEnvironment : IWebhookEnvironment
    {
        private readonly ITwitterClient client;

        public WebhookEnvironment(IWebhookEnvironmentDTO webhookEnvironmentDTO, ITwitterClient client)
        {
            this.client = client;
            WebhookEnvironmentDTO = webhookEnvironmentDTO;
        }

        [JsonIgnore]
        public IWebhookEnvironmentDTO WebhookEnvironmentDTO { get; }

        public string Name => WebhookEnvironmentDTO.Name;

        public IWebhook[] Webhooks => WebhookEnvironmentDTO.Webhooks.Select(x => this.client.Factories.CreateWebhook(x)).ToArray();
    }
}

File: WorldFeed/Client/src/app/core/Core/Models/WebhookEnvironment.ts
import {IWebhook} from '../../Public/Models/Interfaces/IWebhook';
import {IWebhookEnvironment} from "../../Public/Models/Interfaces/IWebhookEnvironment";
import {ITwitterClient} from "../../Public/ITwitterClient";
import {IWebhookEnvironmentDTO} from "../../Public/Models/Interfaces/DTO/Webhooks/IWebhookEnvironmentDTO";

export class WebhookEnvironment implements IWebhookEnvironment {
  private readonly _client: ITwitterClient;

  constructor(webhookEnvironmentDTO: IWebhookEnvironmentDTO, client: ITwitterClient) {
    this._client = client;
    this.webhookEnvironmentDTO = webhookEnvironmentDTO;
  }

  // [JsonIgnore]
  public webhookEnvironmentDTO: IWebhookEnvironmentDTO;

  get name(): string {
    return this.webhookEnvironmentDTO.name;
  }

  get webhooks(): IWebhook[] {
    return this.webhookEnvironmentDTO.webhooks.map(x => this._client.factories.createWebhook(x));
  }
}
